src/main/java/tests/ShopServiceTest.java

```diff
@@ -0,0 +1,110 @@
+package tests;
+
+import core.basesyntax.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ShopServiceTest {
+
+    private ShopService shopService;
+    private List<FruitTransaction> transactions;
+
+    @BeforeEach
+    public void setUp() {
+        Map<Operation, OperationHandler> operationHandlers = new HashMap<>();
+        operationHandlers.put(Operation.BALANCE, new BalanceOperation());
+        operationHandlers.put(Operation.PURCHASE, new PurchaseOperation());
+        operationHandlers.put(Operation.RETURN, new ReturnOperation());
+        operationHandlers.put(Operation.SUPPLY, new SupplyOperation());
+        OperationStrategy operationStrategy = new OperationStrategyImpl(operationHandlers);
+        shopService = new ShopServiceImpl(operationStrategy);
+        transactions = new ArrayList<>();
+    }
+
+    @Test
+    public void test_balance_operation() {
+        FruitTransaction balance = new FruitTransaction();
+        balance.setOperation(Operation.BALANCE);
+        balance.setFruit("banana");
+        balance.setQuantity(100);
+        transactions.add(balance);
+
+        shopService.process(transactions);
+        Map<String, Integer> storage = shopService.getStorage();
+
+        assertEquals(100, (int) storage.get("banana"));
```

Comment:
It's great that you're asserting the state of the storage after a balance operation. However, consider adding more assertions to check if only the expected fruit is affected and no other changes have occurred in the storage.